### DISCUSSION TITLE
Intellisense support for multiple tailwind config files

### DISCUSSION THREAD

**Discussion by thebspin** [tailwindlabs/tailwindcss#9328] [Ideas]
As the title says, i think it would be great if the extension for vscode would be able to support multiple tailwind config files.
Now what is the use case? Well we have a monorepo running using nx for multiple applications and we are adopting tailwind for everyone of these applications.

We write our own plugins to extend Tailwind where it's needed and currently use components to modify certain attributes visually because a button could be rounded in application a but should be square in application b for example.

Now each application already has it's own tailwind config so we figured if we would give our plugin there own configuration we could easily write this plugin once and then setup the config per app so we don't need to build components just to visually modify them. And the sub config's inherit everything from the main config unless it's overwritten.

However this is currently not being picked up by the extension.
This of course will bring a few of it's own problems...
For example when having a plugin that only loads in an app tailwind config means currently the main tailwind config will never know about this and will therefor never suggest autocompletions.

I am not even sure if it would be possible for the extension to know if it a config lives inside a certain directory then it should only show related stuff for files living inside the same directory.

Something that would work in a structure like this:
```
apps
   application-a
      components/
      tailwind.config
   application-b
      components/
      tailwind.config
tailwind.config
```

What are your thoughts?

**Comment by wongjn** [comment]
You could see if the [`tailwindCSS.experimental.configFile`](https://github.com/tailwindlabs/tailwindcss-intellisense#tailwindcssexperimentalconfigfile) setting would fit your needs?

**Reply by thebspin** [reply]
Ah i checked the doc's for something like this but did not read the actually readme of this repo whoops.
So i think this should be the place that makes it works but for now it does not.

I've added to my settings.json:
```
	"tailwindCSS.experimental.configFile": {
		"tailwind.config.js": "**",
		"apps/app-a/tailwind.config.js": "apps/app-a/**"
	}
```

Stuff written in app-a/tailwind.config do work in the final output, when adding or when overwritten.
But the intellisense part does not.
When switching the order above it does pick up on extentions, modifications to the config (or plugins that are only being loaded in that specific config).
In both cases however it does not show the color decorators inline for stuff that's only being defined in the not global config.
